packages/video_player_avplayer/tizen/src/plus_player.cc

```diff
@@ -0,0 +1,666 @@
+
+bool PlusPlayer::SetActivate() {
+  if (!player_->Activate(plusplayer::kTrackTypeVideo)) {
+    LOG_ERROR("[PlusPlayer] Fail to activate video.");
+    return false;
+  }
+  if (!player_->Activate(plusplayer::kTrackTypeAudio)) {
+    LOG_ERROR("[PlusPlayer] Fail to activate audio.");
+    return false;
+  }
+  if (!player_->Activate(plusplayer::kTrackTypeSubtitle)) {
+    LOG_ERROR("[PlusPlayer] Fail to activate subtitle.");
+  }
+
+  return true;
+}
+
+bool PlusPlayer::SetDeactivate() {
+  if (is_prebuffer_mode_) {
+    player_->Stop();
+    return true;
+  }
+
+  if (!player_->Deactivate(plusplayer::kTrackTypeVideo)) {
+    LOG_ERROR("[PlusPlayer] Fail to activate video.");
+    return false;
+  }
+  if (!player_->Deactivate(plusplayer::kTrackTypeAudio)) {
+    LOG_ERROR("[PlusPlayer] Fail to activate audio.");
+    return false;
+  }
+  if (!player_->Deactivate(plusplayer::kTrackTypeSubtitle)) {
+    LOG_ERROR("[PlusPlayer] Fail to activate subtitle.");
+  }
+
+  return true;
+}
+
+void PlusPlayer::Pause() {
+  LOG_INFO("[PlusPlayer] Pause video.");
+  plusplayer::State state = player_->GetState();
+  if (state < plusplayer::State::kReady) {
+    LOG_ERROR("[PlusPlayer] Player is not ready.");
+    return;
+  }
+
+  if (state == plusplayer::State::kPlaying) {
+    if (!player_->Pause()) {
+      LOG_ERROR("[PlusPlayer] Fail to pause video.");
+    }
+  }
+}
+
+void PlusPlayer::SetLooping(bool is_looping) {
+  LOG_ERROR("[PlusPlayer] Not support to set looping.");
+}
+
+void PlusPlayer::SetVolume(double volume) {
+  if (player_->GetState() == plusplayer::State::kPlaying ||
+      player_->GetState() == plusplayer::State::kPaused) {
+    player_->SetVolume(volume);
+  }
+}
+
+void PlusPlayer::SetPlaybackSpeed(double speed) {
+  LOG_INFO("[PlusPlayer] Sets playback speed(%f).", speed);
+  if (player_->GetState() <= plusplayer::State::kIdle) {
+    LOG_ERROR("[PlusPlayer] Player is not prepared.");
+    return;
+  }
+  if (!player_->SetPlaybackRate(speed)) {
+    LOG_ERROR("[PlusPlayer] Fail to set playback rate.");
+  }
+}
+
+void PlusPlayer::SeekTo(int64_t position, SeekCompletedCallback callback) {
+  LOG_INFO("PlusPlayer seeks to position(%lld)", position);
+  if (player_->GetState() < plusplayer::State::kReady) {
+    LOG_ERROR("[PlusPlayer] Player is not ready.");
+    return;
+  }
+
+  if (on_seek_completed_) {
+    LOG_ERROR("[PlusPlayer] Player is already seeking.");
+    return;
+  }
+
+  on_seek_completed_ = std::move(callback);
+  plusplayer::PlayerMemento memento;
+  if (!player_->GetMemento(&memento)) {
+    LOG_ERROR("[PlusPlayer] Fail to get player memento.");
+  }
+
+  if (memento.is_live) {
+    std::string str = player_->GetStreamingProperty("GET_LIVE_DURATION");
+    if (str.empty()) {
+      LOG_ERROR("[PlusPlayer] Fail to get live duration.");
+    }
+    std::vector<std::string> time_str = split(str, '|');
+    int64_t start_time = std::stoll(time_str[0].c_str());
+    int64_t end_time = std::stoll(time_str[1].c_str());
```

Comment:
Possibly out of range if `str` is empty?

Reply:
Fixed.
if `str` is empty, will return `false`.